### PR TITLE
[BREAKING CHANGE] Simplified the release process

### PR DESCRIPTION
**Pull request by wu-sheng**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

* [Breaking Change] All jars of the `skywalking-oap-server` are no longer published through maven central. We will only 
  publish the source tar and binary tar to the website download page, and docker images to docker hub.
  * **Warning** If you are using the `skywalking-oap-server` as a dependency in your project, you need to download the 
    source tar from the website and publish them to your private maven repository.

FYI @apache/skywalking-committers please notice this new proposal. I am not sure who are using these dependencies, but truly, we are not expecting this.

### CODE REVIEW

File: docs/en/guides/How-to-release.md [hunk 1]
```diff
@@ -79,18 +45,18 @@ git push origin v${RELEASE_VERSION}
 ## Stage the release 
 
 ```bash
-./mvnw flatten:flatten install deploy -DskipTests
+./mvnw package -DskipTests
```
Comment (kezhenxu94) [inline docs/en/guides/How-to-release.md:48]: `install` might be still needed if the version is not installed before?

File: docs/en/guides/How-to-release.md [hunk 2]
```diff
@@ -79,18 +45,18 @@ git push origin v${RELEASE_VERSION}
...
 ```
 
 This command will build, sign, and deploy the release to the Apache staging repository.
 You should be prompted for your GPG passphrase during the deployment process.
 If no prompt is present, you can set the passphrase in the environment variable `MAVEN_GPG_PASSPHRASE`:
 
 ```bash
-MAVEN_GPG_PASSPHRASE=<your-gpg-passphrase> ./mvnw flatten:flatten install deploy -DskipTests
```
Comment (kezhenxu94) [inline docs/en/guides/How-to-release.md:90]: GPG related is not needed anymore, it was for maven verification. Now we will only sign the final tar.

**Review by Fine0830 — DISMISSED** [review]

File: tools/releasing/create_source_release.sh [hunk 3]
```diff
@@ -75,5 +75,7 @@ tar czf ${PRODUCT_NAME}-src.tgz \
     ${PRODUCT_NAME}
 
 gpg --armor --detach-sig ${PRODUCT_NAME}-src.tgz
+gpg --armor --detach-sig ${PRODUCT_NAME}.tgz
```
Comment (kezhenxu94) [inline tools/releasing/create_source_release.sh:78]: The file name is `create_source_release.sh`, but these two lines are signing binary package, what do you think moving these two lines to the release doc or renaming this script file?

**Review by kezhenxu94 — COMMENTED** [review]
I did some updates, please check

File: tools/releasing/create_source_release.sh (hunk 3 above)
Comment (wu-sheng) [reply tools/releasing/create_source_release.sh:78]: I prefer rename the script file.

Comment (wu-sheng) [reply tools/releasing/create_source_release.sh]: @kezhenxu94 I updated the docs a little bit more, to move binary into releasing folder.

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**